api/src/main/java/io/strimzi/api/kafka/Crds.java

```diff
@@ -54,84 +62,59 @@ private Crds() {
      * Register custom resource kinds with {@link KubernetesDeserializer} so Fabric8 knows how to deserialize them.
      */
     public static void registerCustomKinds() {
-        for (Class<? extends CustomResource> c : CRDS) {
-            KubernetesDeserializer.registerCustomKind(apiVersion(c), kind(c), c);
+        for (Class<? extends CustomResource> crdClass : CRDS) {
+            for (String version : apiVersion(crdClass)) {
+                KubernetesDeserializer.registerCustomKind(version, kind(crdClass), crdClass);
+            }
         }
     }
 
     private static CustomResourceDefinition crd(Class<? extends CustomResource> cls) {
         String scope;
-        String kind;
         String crdApiVersion;
         String plural;
-        String listKind;
-        String singular;
-        String version;
         String group;
```

Comment:
why all the kind, listKind, singular and version fields are gone?

Reply:
The methods on `Crds` are really intended only to be enough to be able to use our CRDs with Fabric8. (They were never supposed to be sufficiently complete that you could use them to install the CRDs in kube using the Fabric8 API). Fabric8 doesn't need the list kind etc or singular in order to work with its `customResource()` method.